### PR TITLE
feat: Add zoom hotkey polyfill for non windows platforms

### PR DESCRIPTION
**Pull request by Legend-Master**
Reference: #3310, #9352

### CODE REVIEW

File: core/tauri-runtime/src/webview.rs [hunk 1]
```diff
@@ -350,6 +350,13 @@ impl WebviewAttributes {
   }
 
   /// Whether page zooming by hotkeys is enabled
+  ///
+  /// ## Platform-specific:
+  ///
+  /// - **Windows**: Controls WebView2's [`IsZoomControlEnabled`](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2settings?view=webview2-winrt-1.0.2420.47#iszoomcontrolenabled) setting.
+  /// - **MacOS / Linux / Android / iOS**: Requires `webview:allow-set-webview-zoom` permission,
+  /// we'll inject a polyfill that zooms in and out with `ctrl`/`command` + `-`/`=`,
+  /// 20% in each step, ranging from 20% to 1000%.
```
Comment (amrbashir) [inline core/tauri-runtime/src/webview.rs]:
```suggestion
  /// - **MacOS / Linux **: Injects a polyfill that zooms in and out with `ctrl/command` + `-/=`,
  /// 20% in each step, ranging from 20% to 1000%. Requires `webview:allow-set-webview-zoom` permission
  ///
  /// - **Android / iOS **: Unsupported.
```

File: core/tauri/src/webview/mod.rs [hunk 2]
```diff
@@ -776,7 +776,14 @@ fn main() {
...
 
-  /// Whether page zooming by hotkeys is enabled **Windows Only**
+  /// Whether page zooming by hotkeys is enabled
+  ///
+  /// ## Platform-specific:
+  ///
+  /// - **Windows**: Controls WebView2's [`IsZoomControlEnabled`](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2settings?view=webview2-winrt-1.0.2420.47#iszoomcontrolenabled) setting.
+  /// - **MacOS / Linux / Android / iOS**: Requires `webview:allow-set-webview-zoom` permission,
+  /// we'll inject a polyfill that zooms in and out with `ctrl`/`command` + `-`/`=`,
+  /// 20% in each step, ranging from 20% to 1000%.
```
Comment (amrbashir) [inline core/tauri/src/webview/mod.rs]:
```suggestion
  /// - **MacOS / Linux **: Injects a polyfill that zooms in and out with `ctrl/command` + `-/=`,
  /// 20% in each step, ranging from 20% to 1000%. Requires `webview:allow-set-webview-zoom` permission
  ///
  /// - **Android / iOS **: Unsupported.
```

File: core/tauri-utils/src/config.rs [hunk 3]
```diff
@@ -1293,7 +1293,14 @@ pub struct WindowConfig {
...
   pub proxy_url: Option<Url>,
-  /// Whether page zooming by hotkeys is enabled **Windows Only**
+  /// Whether page zooming by hotkeys is enabled
+  ///
+  /// ## Platform-specific:
+  ///
+  /// - **Windows**: Controls WebView2's [`IsZoomControlEnabled`](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2settings?view=webview2-winrt-1.0.2420.47#iszoomcontrolenabled) setting.
+  /// - **MacOS / Linux / Android / iOS**: Requires `webview:allow-set-webview-zoom` permission,
+  /// we'll inject a polyfill that zooms in and out with `ctrl`/`command` + `-`/`=`,
+  /// 20% in each step, ranging from 20% to 1000%.
```
Comment (amrbashir) [inline core/tauri-utils/src/config.rs]:
```suggestion
  /// - **MacOS / Linux **: Injects a polyfill that zooms in and out with `ctrl/command` + `-/=`,
  /// 20% in each step, ranging from 20% to 1000%. Requires `webview:allow-set-webview-zoom` permission
  ///
  /// - **Android / iOS **: Unsupported.
```

File: core/tauri/src/manager/webview.rs [hunk 4]
```diff
@@ -534,6 +534,23 @@ impl<R: Runtime> WebviewManager<R> {
       }
     }
 
+    #[cfg(not(target_os = "windows"))]
```
Comment (amrbashir) [inline core/tauri/src/manager/webview.rs]: Shouldn't be injected on mobile platforms
```suggestion
    #[cfg(all(desktop, not(target_os = "windows")))]
```

File: core/tauri/src/webview/scripts/zoom-hotkey.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+const MAX_ZOOM_LEVEL = 10
+const MIN_ZOOM_LEVEL = 0.2
+
+window.addEventListener('keydown', (event) => {
+  if (event.defaultPrevented) {
+    return
+  }
+  if (OS_NAME === 'macos' ? event.metaKey : event.ctrlKey) {
+    if (event.key === '-') {
```
Comment (amrbashir) [inline core/tauri/src/webview/scripts/zoom-hotkey.js:14]: we should also support `ctrl+0` to reset it to `1`

File: core/tauri/src/webview/webview_window.rs [hunk 6]
```diff
@@ -839,7 +839,14 @@ fn main() {
...
 
-  /// Whether page zooming by hotkeys is enabled **Windows only**
+  /// Whether page zooming by hotkeys is enabled
+  ///
+  /// ## Platform-specific:
+  ///
+  /// - **Windows**: Controls WebView2's [`IsZoomControlEnabled`](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2settings?view=webview2-winrt-1.0.2420.47#iszoomcontrolenabled) setting.
+  /// - **MacOS / Linux / Android / iOS**: Requires `webview:allow-set-webview-zoom` permission,
+  /// we'll inject a polyfill that zooms in and out with `ctrl`/`command` + `-`/`=`,
+  /// 20% in each step, ranging from 20% to 1000%.
```
Comment (amrbashir) [inline core/tauri/src/webview/webview_window.rs]:
```suggestion
  /// - **MacOS / Linux **: Injects a polyfill that zooms in and out with `ctrl/command` + `-/=`,
  /// 20% in each step, ranging from 20% to 1000%. Requires `webview:allow-set-webview-zoom` permission
  ///
  /// - **Android / iOS **: Unsupported.
```

File: core/tauri-runtime/src/webview.rs [hunk 7]
```diff
@@ -350,6 +350,13 @@ impl WebviewAttributes {
   }
 
   /// Whether page zooming by hotkeys is enabled
+  ///
```
Comment (amrbashir) [inline core/tauri-runtime/src/webview.rs:353]: We also need to update the documentation in `tooling/api/webview.ts`

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: core/tauri-runtime/src/webview.rs (hunk 7 above)
Comment (Legend-Master) [reply core/tauri-runtime/src/webview.rs:353]: It only has the zoom API, this shouldn't affect it right?

Comment (amrbashir) [reply core/tauri-runtime/src/webview.rs:353]: it has `zoomHotkeysEnabled` property when creating new webviews

Comment (Legend-Master) [reply core/tauri-runtime/src/webview.rs:353]: Forgot about that, I'll mirror the docs to their as well

**Review by amrbashir — APPROVED** [review]
Thank you

**Pull request merged**